README.md

```diff
@@ -45,171 +45,113 @@ For details on Chronos models, training data and procedures, and experimental re
 
-For the original Chronos models, `pipeline.predict` can be used to draw forecast samples. More options for `predict_kwargs` in `pipeline.predict_quantiles` can be found with:
+# Load future values of covariates
+test_df = pd.read_parquet("s3://autogluon/datasets/timeseries/electricity_price/test.parquet")
+future_df = test_df.drop(columns=target)
 
-```python
-from chronos import ChronosPipeline, ChronosBoltPipeline
-
-print(ChronosPipeline.predict.__doc__)  # for Chronos models
-print(ChronosBoltPipeline.predict.__doc__)  # for Chronos-Bolt models
+# Generate predictions with covariates
+pred_df = pipeline.predict_df(
+    context_df,
+    future_df=future_df,
+    prediction_length=prediction_length,
+    quantile_levels=[0.1, 0.5, 0.9],
+    id_column=id_column,
+    timestamp_column=timestamp_column,
+    target=target,
+)
 ```
 
 We can now visualize the forecast:
 
 ```python
 import matplotlib.pyplot as plt  # requires: pip install matplotlib
 
-forecast_index = range(len(df), len(df) + 12)
-low, median, high = quantiles[0, :, 0], quantiles[0, :, 1], quantiles[0, :, 2]
-
-plt.figure(figsize=(8, 4))
-plt.plot(df["#Passengers"], color="royalblue", label="historical data")
-plt.plot(forecast_index, median, color="tomato", label="median forecast")
-plt.fill_between(forecast_index, low, high, color="tomato", alpha=0.3, label="80% prediction interval")
-plt.legend()
-plt.grid()
-plt.show()
-```
-
-### Extracting Encoder Embeddings
-
-A minimal example showing how to extract encoder embeddings from Chronos models:
-
-```python
-import pandas as pd
-import torch
-from chronos import ChronosPipeline
-
-pipeline = ChronosPipeline.from_pretrained(
-    "amazon/chronos-t5-small",
-    device_map="cuda",
-    torch_dtype=torch.bfloat16,
+timeseries_id = "DE"  # Specific time series to visualize
+history_length = 256  # The number of historical values to plot
+
+ts_context = context_df.query(f"{id_column} == @timeseries_id").set_index(timestamp_column)[target]
+ts_pred = pred_df.query(f"{id_column} == @timeseries_id and target_name == @target").set_index(timestamp_column)[
+    ["0.1", "predictions", "0.9"]
+]
+ts_ground_truth = test_df.query(f"{id_column} == @timeseries_id").set_index(timestamp_column)[target]
+
+start_idx = max(0, len(ts_context) - history_length)
+plot_cutoff = ts_context.index[start_idx]
+ts_context = ts_context[ts_context.index >= plot_cutoff]
+ts_ground_truth = ts_ground_truth[ts_ground_truth.index >= plot_cutoff]
+
+fig = plt.figure(figsize=(12, 3))
+ax = fig.gca()
+ts_context.plot(ax=ax, label=f"historical {target}", color="xkcd:azure")
+ts_ground_truth.plot(ax=ax, label=f"future {target} (ground truth)", color="xkcd:grass green")
+ts_pred["predictions"].plot(ax=ax, label="forecast", color="xkcd:violet")
+ax.fill_between(
+    ts_pred.index,
+    ts_pred["0.1"],
+    ts_pred["0.9"],
+    alpha=0.7,
+    label="prediction interval",
+    color="xkcd:light lavender",
 )
```

Comment:
This could be more concise:
```python
import matplotlib.pyplot as plt

ts_context = context_df.set_index(timestamp_column)[target].tail(256)
ts_pred = pred_df.set_index(timestamp_column)
ts_ground_truth = test_df.set_index(timestamp_column)[target]

ts_context.plot(label=f"historical data", color="xkcd:azure", figsize=(12, 3))
ts_ground_truth.plot(label=f"future data (ground truth)", color="xkcd:grass green")
ts_pred["predictions"].plot(label="forecast", color="xkcd:violet")
plt.fill_between(
    ts_pred.index,
    ts_pred["0.1"],
    ts_pred["0.9"],
    alpha=0.7,
    label="prediction interval",
    color="xkcd:light lavender",
)
plt.legend()
```